core/src/io/anuke/mindustry/world/blocks/power/PowerGraph.java

```diff
@@ -39,37 +38,41 @@ public synchronized void update(){
         lastFrameUpdated = threads.getFrameID();
 
         float totalInput = 0f;
-
         for(Tile producer : producers){
-            totalInput += producer.entity.power.amount;
+            float accumulator = producer.entity.power.amount;
+            if (accumulator > 0.0001f) {
+                totalInput += producer.entity.power.amount;
+            }
         }
 
-        for(Tile producer : producers){
-            float accumulator = producer.entity.power.amount;
+        float maxOutput = 0f;
+        for(Tile consumer : consumers){
+            float canHold = consumer.block().powerCapacity - consumer.entity.power.amount;
+            if (canHold > 0.0001f) {
+                maxOutput += canHold;
+            }
+        }
 
-            if(accumulator <= 0.0001f) continue;
+        if (totalInput == 0 || maxOutput == 0) {
```

Comment:
While this is enough to catch NaN errors, it should be `<= 0.0001f` to catch situations where the input or output is very small and there's not enough power to warrant updating the flow.